cmd/kubeadm/app/phases/certs/certs.go

```diff
@@ -354,8 +354,9 @@ func SharedCertificateExists(cfg *kubeadmapi.ClusterConfiguration) (bool, error)
 }
 
 // UsingExternalCA determines whether the user is relying on an external CA.  We currently implicitly determine this is the case
-// when both the CA Cert and the front proxy CA Cert are present but the CA Key and front proxy CA Key are not.
+// when the CA Cert is present but the CA Key is not.
 // This allows us to, e.g., skip generating certs or not start the csr signing controller.
```

Comment:
as per the above comment, it seems that this change might affect our CSR code?
e.g.:
https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/kubeadm-certs/

but arguably getting HA right is the higher priority.

Reply:
@neolit123 this change should not affect CSR code.

However, being the CSR thing built "improperly" on top of phases and the init workflow, there is a high risk that a change on one will impact the other.

My personal opinion is that in the long term CSR should be moved away because they are not part of the kubeadm init workflow

Reply:
> My personal opinion is that in the long term CSR should be moved away because they are not part of the kubeadm init workflow

absolutely. 
i think there was a feature request about CSR in k/kubeadm, but i must say i mostly missed the demand -> implementation process on this one.